### PR TITLE
paramsNonEmpty returns NonEmptyList[A]

### PR DESCRIPTION
**Pull request by ImLiar**
As it was discussed earlier: https://github.com/finagle/finch/pull/577#issuecomment-202103686

### CODE REVIEW

File: core/src/main/scala/io/finch/Endpoints.scala [hunk 1]
```diff
@@ -337,11 +338,13 @@ trait Endpoints {
    * at least one element) multi-value query-string param `name` from the request into a `Seq` or
    * raises a [[Error.NotPresent]] exception when the params are missing or empty.
    */
-  def paramsNonEmpty(name: String): Endpoint[Seq[String]] =
+  def paramsNonEmpty(name: String): Endpoint[NonEmptyList[String]] =
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/Endpoints.scala]: How do you feel about introducing a new function `paramsNel` and keeping the old one (behind deprecation)? I want to keep the API unchanged and I think `paramsNel` sounds like a reasonable name.

Comment (ImLiar) [inline core/src/main/scala/io/finch/Endpoints.scala]: Okay, let's introduce new method

File: core/src/main/scala/io/finch/Endpoints.scala [hunk 2]
```diff
@@ -357,6 +359,19 @@ trait Endpoints {
     }).shouldNot("be empty")(_.isEmpty)
 
   /**
+   * An evaluating [[Endpoint]] that reads a required multi-value query-string param `name`
+   * from the request into a `NonEmptyList` or raises a [[Error.NotPresent]] exception
+   * when the params are missing or empty.
+   */
+  def paramsNel(name: String): Endpoint[NonEmptyList[String]] =
+    option(items.ParamItem(name))(requestParams(name)).mapAsync({
+      case values if values.exists(_.nonEmpty) =>
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/Endpoints.scala]: Can we simplify that and perform just a single traversal? Something like:

```scala
.mapAsync { values => values.filter(_.nonEmpty) match {
  case hd :: tl => Future.value(NonEmptyList(hd, tl))
  case _ => Future.exception(Error.NotPresent(items.ParamItem(name)))
}}
```

Comment (ImLiar) [inline core/src/main/scala/io/finch/Endpoints.scala]: Emh, but in that case it could contain empty parameters

Comment (ImLiar) [inline core/src/main/scala/io/finch/Endpoints.scala]: Ah, I see.

File: examples/src/main/scala/io/finch/div/Main.scala [hunk 3]
```diff
@@ -24,6 +23,10 @@ import shapeless._
  */
 object Main extends App {
 
+  (1,2,3) match {
```
Comment (vkostyukov) [inline examples/src/main/scala/io/finch/div/Main.scala]: Some unwanted changes.

Comment (ImLiar) [inline examples/src/main/scala/io/finch/div/Main.scala]: I should go sleep :(

**Comment by vkostyukov** [comment]
Thanks @ImLiar! Merging this!

**Pull request merged**